토큰 생성/갱신 API 및 restDocs 의존성 추가

olivejua opened the pull request:
## 내용
*  토큰 API 생성/갱신/삭제 API 생성
* Refresh Token Entity&Repository 생성
* RestDocs 의존성 추가 및 통합테스트 API 요청시 RestDocs snippet 추가 구현
* 루트 프로젝트의 test환경에 MockDomain 객체 추가

olivejua committed:
chore: restdocs 의존성 및 플러그인 추가

olivejua committed:
test: restdocs 테스트애노테이션 추가

olivejua committed:
feat: 현재 인증되어있는 로그인한 사용자 정보 조회 구현

olivejua committed:
refactor: 패키지명 변경 user/service -> user/business

olivejua committed:
feat: RefreshToken DB table 추가 및 repository, service 발급 구현

olivejua committed:
refactor: 클래스명 변경 RefreshTokenService -> RefreshTokenDomainService

olivejua committed:
feat: 엑세스, 리프레시 토큰 발급 API 구현

olivejua committed:
test: Mock 객체 루트프로젝트에 생성

olivejua committed:
feat: loginId로 App, Social Account 찾아오기 구현

olivejua committed:
feat: App, Social 계정 인증 구현

olivejua committed:
test: Mock Test 클래스 import 문 정리

olivejua committed:
refactor: SignInRequest 제거

- 사용안함

olivejua committed:
feat: 로그인 API 생성 및 로그인 인증 구현

olivejua committed:
refactor: 토큰 갱신 Response 타입 변경

olivejua committed:
test: 토큰 갱신 Test Success case 구현

olivejua committed:
feat: 리프레시 토큰 제거 구현

olivejua committed:
refactor: 중복된 의존성 제거

olivejua committed:
test: restdocs index 페이지 생성 및 common 스니펫 테스트 케이스 작성

olivejua committed:
refactor: common 모듈 안 service 구조 변경

f-lab-sean commented on auth/src/main/java/com/food/auth/business/LoginService.java:
```diff
@@ -0,0 +1,71 @@
+                .accessToken(accessTokenProvider.create(user.getId()))
+                .refreshToken(refreshTokenService.create(user.getId()))
+                .userId(user.getId())
+                .build();
```
로그인 성공시 응답 필드에 loginId, nickname 정도는 있어야 화면에 표시할 수 있겠네요.

f-lab-sean replied:
소셜로그인의 경우 LoginId 만을 가지고 하는데,
소셜로그인을 구현한다면 이와 다른 방식의 로그인 처리 과정이 필요합니다.

실제 소셜의 Resource Server로부터 인증을 통한 AccessToken 발행이 필요합니다.
이후에 이 토큰을 활용해서 소셜로부터 사용자 정보를 가져와서 활용할 수 있습니다.

소셜 사용자에 대한 고민이 좀 더 필요해 보입니다.

olivejua replied:
조금 더 공부하여 수정해서 올리겠습니다!

f-lab-sean commented on auth/src/main/java/com/food/auth/business/LoginService.java:
```diff
@@ -0,0 +1,71 @@
+    public void logout(Long userId) {
+        refreshTokenService.delete(userId);
+    }
+
```
사용자 로그인/로그아웃에 대한 이력은 남기는 게 좋을 것 같습니다.

olivejua replied:
로그인 이력은 최근 활동여부를 확인하는 용도라면  
로그아웃 이력은 남기는 이유가 인증 요청시 리프레시 토큰을 DB에서 찾을 수 없을 경우에 대한 이슈 추적 용도인가요?🙂   
(검색해도 잘 찾을 수 없어 여쭤봅니다! ㅠㅠ)

f-lab-sean replied:
슬랙에만 답을 했네요. ;
보통 앱이나 서비스에 최근 로그아웃 정보를 보여줍니다. 비정상적인 로그인/로그아웃을 추후에 추적할 때 사용합니다.

f-lab-sean commented on common/src/main/java/com/food/common/user/business/dto/response/accountDomain/FoundAppAccount.java:
```diff
@@ -15,4 +15,8 @@ public FoundAppAccount(AppAccount account, FoundUser user) {
     }
+
+    public boolean matchesPassword(String password) {
+        return this.password.equals(password);
```
패스워드는 기본적으로 단방향 암호화가 되어야 하므로
회원 등록 구현할 때 같이 수정이 필요합니다.

olivejua replied:
넵 회원가입시 비밀번호는 Bcrypt로 암호화하여 DB 저장할 예정입니다!

olivejua committed:
refactor: 로그인 response에 nickname, loginId 추가

olivejua committed:
Merge pull request #17 from f-lab-edu/feature/refactor-services-in-common

common 모듈 service return type 변경

olivejua committed:
Merge pull request #15 from f-lab-edu/feature/remove-dependencies-duplicated

중복된 의존성 제거

olivejua committed:
Merge pull request #16 from f-lab-edu/feature/create-indexfile-of-api-docs

restdocs index 페이지 생성 및 common 스니펫 테스트 케이스 작성

olivejua committed:
Merge branch 'feature/issue-auth-token' of https://github.com/f-lab-edu/food-service into feature/issue-auth-token

 Conflicts:
	auth/src/main/java/com/food/auth/presentation/dto/TokenIssueResponse.java
	user/src/main/java/com/food/user/presentation/AccountController.java

olivejua committed:
refactor: 로그인 response에 nickname, loginId 추가 및 api 문서 설명 추가

olivejua committed:
refactor: 로그인/로그아웃 로그 생성

- LogInOutLog Entity, Repository 생성
- LogInOutLog 데이터 저장 CommonService 생성
- auth service에서 로그인, 로그아웃시 log 저장 함수 호출 추가
